### PR TITLE
Function to detect Participant analysis failure and abort the Group analysis

### PR DESCRIPTION
**Pull request by tushargupta98**
I have added function which detect failure in a participant analysis and aborts the group analysis with an error message.

### CODE REVIEW

File: sim/SparkBIDS.py [hunk 1]
```diff
@@ -36,48 +36,54 @@ def run(self, sc):
...
+            self.participant_analysis_failed=False
             for result in mapped.collect():
                 self.pretty_print(result)
+                if self.check_failure(result):
+                    #Disable Group Analysis if Participant Analysis Fails
+                    self.do_group_analysis=False
+                    self.participant_analysis_failed=True
 
+        if self.participant_analysis_failed:
+            print("Error# Participant analysis failed. Aborting group analysis")
```
Comment (glatard) [inline sim/SparkBIDS.py:51]: you can print this message in the previous if block (line 48). Then you don't need `self.participant_analysis_failed`. Print the message only if `self.do_group_analysis`.

**Review by glatard — CHANGES_REQUESTED** [review]
Great start! See (minor) comment.

**Comment by glatard** [comment]
Also, the PR has to be done against the `develop` branch, not `master`.

**Comment by tushargupta98** [comment]
Sure Professor,
I will create a new PR for Develop branch and cancel the old one, if
possible.
For the previous comment, when I was making the change I wasn't sure as how
many times the error can occur, I mean to say whether it will happen just
once for each participant or if there is a different mechanism. Also, I
thought of consolidating the failure for both the participants into the
one  error message.

I will make the necessary change in the develop branch and create a new PR.

Thanks and Regards,
Tushar

**Pull request closed without merging**